# 1 - Récupération du fichier

In [1]:
import pandas as pd

In [14]:
musees = pd.read_csv("liste-des-musees-de-france-2021.csv", sep=";", encoding="latin-1")

In [60]:
#Aperçu du dataframe :
musees.head()

osm_id                         name number                      street  \
0  4540234591                      3 bis f    109  Avenue du Petit Barthélémy   
1  4388745689             A casa di Losari    NaN                        T 30   
2  4426176194             À l'homme volant     35                  Grande Rue   
3  7652598615  Abbadiale - Maison des arts    NaN              Rue du Gabizos   
4   121580330                       Abbaye    NaN             Rue de l'Abbaye   

   postal_code              city country country_code        lat       lon  \
0      13617.0   Aix-en-Provence  France           fr  43.521207  5.439875   
1      20226.0         Belgodère  France           fr  42.638732  9.007660   
2      72300.0  Sablé-sur-Sarthe  France           fr  47.839656 -0.334043   
3      65400.0  Arras-en-Lavedan  France           fr  42.990826 -0.128057   
4      39210.0    Château-Chalon  France           fr  46.753866  5.624296   

                     website              email         phone  fax  \
0       http://www.3bisf.com  contact@3bisf.com  3.344216e+10  NaN   
1                        NaN                NaN           NaN  NaN   
2                        NaN                NaN  3.324395e+10  NaN   
3  https://www.abbadiale.fr/                NaN  3.363317e+10  NaN   
4                        NaN                NaN           NaN  NaN   

                                      tags description        date_added  \
0          osm:museum;art:art contemporain         NaN  44116,0076388889   
1                               osm:museum         NaN  43338,9393981481   
2  osm:museum;type:centre d'interpretation         NaN  42644,4636574074   
3                               osm:museum         NaN  44120,0466782407   
4              osm:museum;art:architecture         NaN  43102,5491898148   

  wikidata  
0      NaN  
1      NaN  
2      NaN  
3      NaN  
4      NaN

# 2- Réponse aux questions

In [17]:
print("Il y a au total " + str(len(musees)) + " musées en France métropolitaine")

Il y a au total 3784 musées en France métropolitaine


In [35]:
musees['city'].value_counts()

Paris               175
Lyon                 33
Marseille            28
Grenoble             22
Bordeaux             20
                   ... 
Jarnac                1
Tréguier              1
Pessac                1
Belmont               1
Saint-Cyr-en-Val      1
Name: city, Length: 2295, dtype: int64

#### C'est à Paris qu'il y a le plus de musées

## Pour les deux questions qui suivent je ne considèrerai que le nombre de villes possédant un musée

In [37]:
round(musees['city'].value_counts().mean(),1)

1.6

#### Il y a en moyenne 1,6 musée par ville


In [38]:
round(musees['city'].value_counts().median(),1)

1.0

#### La médiane est 1 musée par ville

In [48]:
types_musees = musees['tags'].str.split(";")

musees_sorted = types_musees.explode().value_counts()

In [55]:
musees_percent = (musees_sorted / musees_sorted[0])*100

# Si l'on prend la totalité des musées, voici la répartition des types de musées pour les 30 premiers (certains n'ont que le type "musée")
musees_percent.drop('osm:museum').head(40)

type:ecomusee                         10.505425
type:musee technique et industriel     5.557026
type:musee d'histoire                  4.260386
gender:musee d'homme                   3.757608
art:architecture                       3.334215
type:musee de beaux-arts               3.307753
type:musee d'art                       2.884361
type:musee archeologique               2.884361
art:prehistoire                        2.619741
art:art contemporain                   1.720032
type:maison d'artiste                  1.720032
type:musee d'arts populaires           1.534798
type:musee d'histoire naturelle        1.508336
type:centre d'interpretation           1.217253
type:musee litteraire                  0.873247
musee d'histoire                       0.661551
type:musee de sciences                 0.608627
type:centre d'art                      0.529241
gender:musee de femme                  0.476316
type:fondation                         0.476316
art:antiquite                          0

In [59]:
musees_sorted2 = musees_sorted.drop('osm:museum')
musees_percent2 = (musees_sorted2 / musees_sorted2.sum())*100

#Si l'on prend uniquement les musées ayant une catégorie autre que "musée" renseignée, voici les pourcentages des 30 premiers
musees_percent2.head(30)

type:ecomusee                         19.547021
type:musee technique et industriel    10.339734
type:musee d'histoire                  7.927129
gender:musee d'homme                   6.991630
art:architecture                       6.203840
type:musee de beaux-arts               6.154604
type:musee d'art                       5.366814
type:musee archeologique               5.366814
art:prehistoire                        4.874446
art:art contemporain                   3.200394
type:maison d'artiste                  3.200394
type:musee d'arts populaires           2.855736
type:musee d'histoire naturelle        2.806499
type:centre d'interpretation           2.264894
type:musee litteraire                  1.624815
musee d'histoire                       1.230921
type:musee de sciences                 1.132447
type:centre d'art                      0.984737
gender:musee de femme                  0.886263
type:fondation                         0.886263
art:antiquite                          0

In [63]:
musees["name"].str.startswith('Château').value_counts().loc[True]

57

#### Il y a 57 musées dont le nom commence par Château

In [74]:
len(musees["website"].dropna())

1636

#### Nous avons les sites web pour 1636 musées

In [91]:
div_postal_code = musees['postal_code'].dropna()/1000
dpt_code = pd.DataFrame ([int(code) for code in div_postal_code], columns =['code_postal'])
dpt_code.value_counts()

code_postal
75             172
59              96
33              93
6               87
7               83
              ... 
23              10
90               9
48               8
98               8
590              1
Length: 97, dtype: int64

#### Le département avec le plus de musées est le 75 (Paris) avec 172 musées. Le 48 (Lozère) et le 98 (erreur..? pas de département 98 en France métropolitaine) ne possèdent chacun que 8 musées. Il y a une faute de frappe sur le 590 :)

In [95]:
musees["name"].str.contains('Napoléon|Napoleon').value_counts().loc[True]

5

#### Il y a 5 musées dont le nom contient 'Napoléon' (avec ou sans accent)

# 3 - Pour aller plus loin

Plusieurs informations pourraient être intéressantes à récupérer :
- les notations Google / Tripadvisor pour donner une idée de la qualité du musée. Source: Google/Tripadvisor
- le statut du musée et les éventuelles gratuités (premier dimanche du mois, étudiants UE, etc..). Source : site du musée ? ou un statut existe déjà qui regroupe tout ça ? Je ne suis pas assez calée :)